### DISCUSSION TITLE
How can I tell if islanding is in effect / grid is down?

### DISCUSSION THREAD

**Discussion by konstar** [WillCodeForCats/solaredge-modbus-multi#281] [Q&A]
Hi,

Does SolarEdge expose whether the grid is down and/or that my system has become an island?

I thought about checking Grid export / import values, but guessing that would lead to many false positives.

Once I can detect this event, I can automate keeping my load down to essential tasks only, in order to prolong battery storage during a grid outage.
 
thanks.

**Comment by cralwalker** [comment]
I have a backup interface BI-AU1P on the wired Modbus if this turns into dev that need testing.

**Reply by WillCodeForCats** [reply]
There isn't anything for this device in modbus that I'm aware of.

**Reply by konstar** [reply]
Thanks guys. Any thoughts on an alternative way to discover when the grid is offline?

**Comment by konstar** [comment]
What does M_EVENT_Power_Failure signal? How would I confirm what it means? The docs I've seen just state the flag and enum values.

I did some offline testing a few weeks ago and didn't notice any events come through. So guessing it is a dead end.

Another possibility from the Sunspace standard is GRID_DISCONNECT (0x4) on Evt.
The SolarEdge standard doesn't seem to reference it (see Global Events in https://knowledge-center.solaredge.com/sites/kc/files/sunspec-implementation-technical-note.pdf), but the SunSpec standard does. Is there a way I could trace Evt to see if this flag comes through when I pull the power?

Thanks

SunSpec Reference: https://github.com/sunspec/models

**Reply by WillCodeForCats** [reply]
The global events sensor is already supported but currently only appears as an entity for Synergy Technology model inverters, which are typically the three phase commercial models from SolarEdge. I could make it always available even if the inverter indicates it's not supported, but I don't know that it will give any values on a non-Synergy platform.

SunSpec contains a significant amount of things that SolarEdge simply does not implement.

**Reply by WillCodeForCats** [reply]
> I could make it always available even if the inverter indicates it's not supported, but I don't know that it will give any values on a non-Synergy platform.

This is actually impossible but I didn't remember until I looked at the code. SolarEdge reuses an address for different purposes depending on if the inverter is a synergy model or not. If you have a synergy model inverter then the address used for global events is HR40129. But if you have a non-synergy inverter then HR40129 is used for Meter 1.

So unless you have a synergy inverter you can't get those event flags.
